Tests/Sources/VLCLibraryTest.swift

```diff
@@ -0,0 +1,119 @@
+/*****************************************************************************
+ * VLCLibraryTest.swift
+ *****************************************************************************
+ * Copyright (C) 2018 VLC authors and VideoLAN
+ * $Id$
+ *
+ * Authors: Mike JS. Choi <mkchoi212 # icloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+
+import XCTest
+
+class VLCLibraryTest: XCTestCase {
+    
+    let paramKey = "VLCParams"
+    
+    override func setUp() {
+        super.setUp()
+        UserDefaults.standard.removeObject(forKey: paramKey)
+    }
+
+    func testSharedLibrary() {
+        XCTAssertNotNil(VLCLibrary.shared())
+        XCTAssertNotNil(VLCLibrary.shared().instance)
+    }
+    
+    func testInitWithOptions() throws {
+        let tests = [
+            ["--no-video-title-show", "--verbose=4"],
+            []
+        ]
+        
+        for option in tests {
+            let library = try XCTAssertNotNilAndUnwrap(VLCLibrary(options: option))
+            XCTAssertNotNil(library.instance)
+            assertDefaultParameters()
```

Comment:
Wouldn't it be easier to understand, recognize why a test fails if you don't have a for loop and dict but write the params to test inline?
I know it's less code but right now I have to take some time to understand what line corresponds to a failing test if it would fail here. I'm also a bit unsure why we need to assertDefaultParams for : ["--no-video-title-show", "--verbose=4"],

Reply:
There is no way to actually see if the custom options were successfully saved because the options are saved via 

```objc
_instance = libvlc_new(count, lib_vlc_params);
```

where `_instance` is a `void *`.

So I just ended up making two dummy options and checking if the library was initialized successfully.

Reply:
Maybe I could init a mock `libvlc_new(count, lib_vlc_params)` and compare the two instances?? Is there a `libvlc_compare_library` method?

Reply:
I don't think there is but you can cast or bridge the void pointer into the correct object :)